### PR TITLE
[Merged by Bors] - chore(Algebra/TrivSqZeroExt): use `open scoped RightActions`

### PR DESCRIPTION
**Pull request by eric-wieser**
One lemma statement has changed (up to associativity).

This is one of the more compelling justifications for #8909.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/TrivSqZeroExt.lean [hunk 1]
```diff
@@ -615,20 +615,20 @@ theorem fst_pow [Monoid R] [AddMonoid M] [DistribMulAction R M] [DistribMulActio
 
 theorem snd_pow_eq_sum [Monoid R] [AddMonoid M] [DistribMulAction R M] [DistribMulAction Rᵐᵒᵖ M]
     (x : tsze R M) (n : ℕ) :
-    snd (x ^ n) = ((List.range n).map fun i => x.fst ^ (n.pred - i) • op (x.fst ^ i) • x.snd).sum :=
+    snd (x ^ n) = ((List.range n).map fun i => x.fst ^ (n.pred - i) •> x.snd <• x.fst ^ i).sum :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/TrivSqZeroExt.lean:618]: Note that this statement now has different associativity compared to what it was before.

**Review by j-loreaux — COMMENTED** [review]
This is indeed much nicer. I went through the file and saw a few places where I would prefer to change `•` to `•>` to aid the eye. I'm only interested in it in places where there is an op smul. Personally, I would like to prefer that as the default style (i.e., if a `SMul Rᵐᵒᵖ M` instance is in context, we write `•>` instead of `•`).

Because GitHub doesn't allow me to easily reference things not in the diff, I'll just list the line numbers: 526, 578, 624, 636, 646, 769.

File: Mathlib/Algebra/TrivSqZeroExt.lean (hunk 1 above)
Comment (j-loreaux) [reply Mathlib/Algebra/TrivSqZeroExt.lean:618]: Thanks, but I'm not concerned about this change. It's minor, and I think the precedences for the `•>` and `<•` notations are good, so maybe this should be considered the more natural version.

Comment (eric-wieser) [reply Mathlib/Algebra/TrivSqZeroExt.lean:618]: I agree that it's minor, which is why I made it anyway :)

**Comment by eric-wieser** [comment]
> Because GitHub doesn't allow me to easily reference things not in the diff, I'll just list the line numbers:

1. https://github.com/leanprover-community/mathlib4/blob/a0ebb82fdb38eb8a2ab4bfafca762955031ae6d2/Mathlib/Algebra/TrivSqZeroExt.lean#L526

    Slightly awkward because I have to then use `op_one` in the proof before I can use `one_smul`

1. https://github.com/leanprover-community/mathlib4/blob/a0ebb82fdb38eb8a2ab4bfafca762955031ae6d2/Mathlib/Algebra/TrivSqZeroExt.lean#L578

    Same, but for `0`

1. https://github.com/leanprover-community/mathlib4/blob/a0ebb82fdb38eb8a2ab4bfafca762955031ae6d2/Mathlib/Algebra/TrivSqZeroExt.lean#L624

    https://github.com/leanprover-community/mathlib4/blob/a0ebb82fdb38eb8a2ab4bfafca762955031ae6d2/Mathlib/Algebra/TrivSqZeroExt.lean#L636

    In this case the point of the lemma is that the two actions are in this case the same; but maybe that's not so good an argument.

1. https://github.com/leanprover-community/mathlib4/blob/a0ebb82fdb38eb8a2ab4bfafca762955031ae6d2/Mathlib/Algebra/TrivSqZeroExt.lean#L646

    I think the argument above applies more strongly here; if you have `IsCentralScalar` you never need to make the distinction

1. https://github.com/leanprover-community/mathlib4/blob/a0ebb82fdb38eb8a2ab4bfafca762955031ae6d2/Mathlib/Algebra/TrivSqZeroExt.lean#L769

    This is an action by `S` not `R`, which has no right action clearly I should change the variable name

**Comment by eric-wieser** [comment]
> (i.e., if a `SMul Rᵐᵒᵖ M` instance is in context, we write `•>` instead of `•`).

It would be neat if we could teach the pretty-printer this rule.

**Comment by eric-wieser** [comment]
I've addressed 1 and 2 above, and 5 moot by renaming. I've solved 3 by adding a second lemma stated with right actions, and I claim 4 needs no change.

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**